src/saturn_engine/core/message.py

```diff
@@ -1,6 +1,12 @@
 import dataclasses
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
 
 
 @dataclasses.dataclass
 class Message:
-    data: str
+    body: str
+
+    @asynccontextmanager
+    async def process(self) -> AsyncIterator:
```

Comment:
Why is this a context manager?

Reply:
so that we can do

```
async with message.process():
   execute_task(message)
```

See `queues/rabbitmq.py` for an exemple.

A different queue system could wrap the message such that:

```
def process(self):
  try:
    yield
    self.queue.ack(self.id)
  except:
    self.queue.nack(self.id)
```